### PR TITLE
CLN: Rename "add" to "sum" in groupby

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

sum is the one method whose name doesn't agree with the corresponding `how`

### CODE REVIEW

**Comment by attack68** [comment]
When looking at this it reminded me of Pokemon: 'are you sure you caught them all?'

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1020,6 +1020,10 @@ def curried(x):
                 return self.apply(curried)
 
             is_transform = name in base.transformation_kernels
+
+            if is_transform and self._obj_with_exclusions.empty:
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:1024]: Was this part of testing https://github.com/pandas-dev/pandas/pull/47840? i.e. should this change be here?

Comment (rhshadrach) [reply pandas/core/groupby/groupby.py]: Yea - that's exactly it. Thanks.

**Review by mroeschke — APPROVED** [review]

**Comment by rhshadrach** [comment]
@attack68 - I think our test coverage is sufficient to feel good I got all the code changes. For comments, I would guess it's possible I missed some. I used `grep -nr --include \*.p* "add"` in both the groupby and the _libs directories and went through the results (and just double checked now).

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**